marklogic-data-hub/src/main/java/com/marklogic/hub/impl/HubProjectImpl.java

```diff
@@ -543,6 +542,60 @@ public void upgradeProject() throws IOException {
         removeEmptyRangeElementIndexArrayFromFinalDatabaseFile();
     }
 
+    @Override
+    public void exportProject() throws IOException {
+        File exportDir = getProjectDir().resolve("build").toFile();
+        if(!exportDir.exists()) {
+            exportDir.mkdirs();
+        }
+        String exportFile  = exportDir.getAbsolutePath() + File.separator + "datahub-project.zip";
+
+        ZipOutputStream zout = new ZipOutputStream(new FileOutputStream(exportFile));
+        List<String> whiteList = new ArrayList<>(Arrays.asList("entities", "flows", "src" + File.separator + "main",
+            "step-definitions", "gradle", "gradlew", "gradlew.bat", "build.gradle", "gradle.properties"));
+        whiteList.forEach(file ->{
+            File fileToZip = getProjectDir().resolve(file).toFile();
+            try{
+                if(fileToZip.isDirectory()) {
+                    zout.putNextEntry(new ZipEntry(file + File.separator));
+                    zipSubDirectory(file + File.separator, fileToZip, zout);
+                }
+                else {
+                    zipSubDirectory("" , fileToZip, zout);
+                }
+            }
+            catch (Exception e){
+                logger.error("Unable to export project", e.getMessage() );
```

Comment:
We definitely don't want to catch and swallow any exception - this could result in some of the zip being written but not all of it. We instead want every exception to be handled as described above.